mne/beamformer/_compute_beamformer.py

```diff
@@ -239,3 +239,187 @@ def _prepare_beamformer_input(info, forward, label, picks, pick_ori,
                          'values are: "dipole", "vertex" or None.')
 
     return is_free_ori, ch_names, proj, vertno, G
+
+
+def _compute_beamformer(beamformer, G, Cm, reg, n_orient, weight_norm,
+                        pick_ori, reduce_rank, rank, is_free_ori,
+                        inversion=None):
+    """Compute a spatial filter (LCMV or DICS)."""
+    # Tikhonov regularization using reg parameter d to control for
+    # trade-off between spatial resolution and noise sensitivity
+    if beamformer == 'lcmv':
+        Cm_inv, d = _reg_pinv(Cm.copy(), reg)
+    elif beamformer == 'dics':
+        Cm_inv, _ = _reg_pinv(Cm, reg, rcond='auto')
+
+    if weight_norm is not None and inversion is not 'single':
+        # Compute square of Cm_inv used for weight normalization
+        Cm_inv_sq = np.dot(Cm_inv, Cm_inv)
+
+        if weight_norm == 'nai':
+            # estimate noise level based on covariance matrix, taking the
+            # smallest eigenvalue that is not zero
+            noise, _ = linalg.eigh(Cm)
+            if rank is not None:
+                rank_Cm = rank
+            else:
+                rank_Cm = estimate_rank(Cm, tol='auto', norm=False,
+                                        return_singular=False)
+                noise = noise[len(noise) - rank_Cm]
+
+                # use either noise floor or regularization parameter d
+                noise = max(noise, d)
+
+    # compute spatial filter
+    W = np.dot(G.T, Cm_inv)
+    n_sources = G.shape[1] // n_orient
+
+    for k in range(n_sources):
+        Wk = W[n_orient * k: n_orient * k + n_orient]
+        Gk = G[:, n_orient * k: n_orient * k + n_orient]
+        if beamformer == 'lcmv' and np.all(Gk == 0.):
+            continue
+        Ck = np.dot(Wk, Gk)
+
+        if beamformer == 'dics':
+            # Normalize the spatial filters:
+            if Wk.ndim == 2 and len(Wk) > 1:
+                # Free source orientation
+                if inversion == 'single':
+                    # Invert for each dipole separately using plain division
+                    Wk /= np.diag(Ck)[:, np.newaxis]
+                elif inversion == 'matrix':
+                    # Invert for all dipoles simultaneously using matrix
+                    # inversion.
+                    Wk[:] = np.dot(linalg.pinv(Ck, 0.1), Wk)
+            else:
+                # Fixed source orientation
+                Wk /= Ck
```

Comment:
What do you mean exactly? Line 297 is triggered when the forward model is in "fixed" mode: each source point only has a single dipole defined: there cannot be any orientation picking.

Reply:
Or do you mean the entire code block starting from line 284?

Reply:
Yes, I mean the whole block (from ``# Normalize the spatial filters:``). I am wondering whether it might have implications for the orientation picking (``pick_ori='max-power'``), doing it one way or the other.

Reply:
You're right! That was not intentional. Good work @britta-wstnr!

Reply:
what's the way forward here?

Reply:
@wmvanvliet Note that for LCMV we do the actual computation of the weights (DICS code above) only at the end, after the (potential) orientation selection etc. Can you confirm that this would work for the DICS case, too? -- As far as I see it, the unit-gain / no orientation selection case would not be affected by this anyway.
With your confirmation, I would incorporate your DICS options (``inversion``) into the LCMV code block at the end of the function.